### DISCUSSION TITLE
Editor Pattern Presets idea

### DISCUSSION THREAD

**Discussion by Paladynee** [ppy/osu#20851] [Ideas] [answered]
I had this idea for an addition to the editor: pattern presets.

**How they are stored:** Pattern presets are saved in the local computer. They persist between beatmaps and beatmap diffs.
**How to make pattern presets:** ou make a pattern/slider in the editor, select it and right click > add as preset.

Because BPM is not a constant thing and might change between songs, osu stores time between objects as beats.
By default, a preset is 1/4 but this can be changed. Non-snapped/offtimed objects are rounded to the closest pattern divisor. This might cause 2b patterns.

**How to use them:** Presets are drag&dropped into the editor. The beat divisor of a pattern and its X/Y size or rotation can be changed freely. after hitting the APPLY button, patterns are converted into beatmap hitobjects. 

a pattern saved as 1/8, placed as 1/4 advances 2x slower.
a pattern saved as 1/4, placed as 1/4 advances normally.
a pattern saved as 1/4, placed as 1/8 advances 2x faster.

![image](https://user-images.githubusercontent.com/62181440/197218429-39b37ef9-f679-428e-a428-f22799dd10ec.png)

**Comment by frenzibyte** [comment, marked as answer]
https://github.com/ppy/osu/issues/8773
